src/test/java/org/broadinstitute/consent/http/resources/ConsentVoteResourceTest.java

```diff
@@ -31,30 +37,179 @@
 
     private ConsentVoteResource resource;
 
+    private Vote createMockVote() {
+        Vote vote = new Vote();
+        vote.setVoteId(RandomUtils.nextInt(100, 1000));
+        vote.setVote(false);
+        vote.setRationale("Test");
+        return vote;
+    }
+
+    private Consent createMockConsent() {
+        Consent consent = new Consent();
+        consent.setConsentId(UUID.randomUUID().toString());
+        return consent;
+    }
+
+    private class LogHandler extends Handler {
+        Level lastLevel = Level.FINEST;
+
+        public Level checkLevel() {
+            return lastLevel;
+        }
+
+        public void publish(LogRecord record) {
+            lastLevel = record.getLevel();
+        }
+
+        public void close(){}
+        public void flush(){}
+    }
+
     @Before
     public void setUp() {
-        MockitoAnnotations.initMocks(this);
+        openMocks(this);
     }
 
     private void initResource() {
         resource = new ConsentVoteResource(emailNotifierService, electionService, voteService);
     }
 
     @Test
-    public void testCreateConsentVote() throws Exception {
-        Consent consent = new Consent();
-        consent.setConsentId(UUID.randomUUID().toString());
-        Vote vote = new Vote();
-        vote.setVoteId(RandomUtils.nextInt(100, 1000));
-        vote.setVote(false);
-        vote.setRationale("Test");
+    public void testCreateConsentVoteSuccess() throws Exception {
+        Vote vote = createMockVote();
+        Consent consent = createMockConsent();
+
         when(voteService.updateVoteById(any(), any())).thenReturn(vote);
         when(electionService.validateCollectEmailCondition(any())).thenReturn(true);
         doNothing().when(emailNotifierService).sendCollectMessage(any());
         initResource();
 
-        Response response = resource.firstVoteUpdate(vote, UUID.randomUUID().toString(), vote.getVoteId());
+        Response response = resource.firstVoteUpdate(vote, consent.getConsentId(), vote.getVoteId());
+        assertEquals(200, response.getStatus());
+    }
+
+    @Test
+    public void testCreateConsentVoteCollectMessageError() throws Exception {
+        Vote vote = createMockVote();
+        Consent consent = createMockConsent();
+
+        Logger logger = Logger.getLogger(ConsentVoteResource.class.getName());
+        LogHandler handler = new LogHandler();
+        handler.setLevel(Level.ALL);
+        logger.setUseParentHandlers(false);
+        logger.addHandler(handler);
+        logger.setLevel(Level.ALL);
```

Comment:
This is very nice. It would be super helpful to extract this into a default interface so other test classes can use the same approach. Would be happy to pair with you on this suggestion, but you can see a similar pattern in `WithMockServer`. That interface has default methods that can be called by any class that implements it. In this case, we might have a default method `default LogHandler setLoggerForClass(String className) { ... blah ... }` and then you can do the assert below with log handler. Of course, there are other ways of implementing this, feel free to explore as you see fit.

Reply:
Good idea! Would you like to add this as a story to Jira or should I?

Reply:
I created DUOS-1515 for this.